lib/suseconnect.pm

```diff
@@ -0,0 +1,107 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Register base product & extensions with SUSEConnect
+# Maintainer: Jiawei Sun <jwsun@suse.com>
+
+package suseconnect;
+
+use base Exporter;
+use Exporter;
+
+use strict;
+use testapi;
+
+
+our @EXPORT = qw (suseconnect_registeration command_register);
+
+sub suseconnect_registeration {
+    my $product_version = get_var('VERSION');
+
+    # register base prodcut
+    command_register($product_version);
+    my @scc_addons = split(/,/, get_var('SCC_ADDONS', ''));
+
+    # register extensions without reg code
+    for my $addon (@scc_addons) {
+        next unless grep { $addon eq $_ } qw(asmm contm lgm pcm tcm wsm);
```

Comment:
We already have all this information in lib/registration.pm and not only this but also much more other code. I strongly suggest to not duplicate this. Please 
[DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). You are calling the same code multiple times, please [extract method](http://refactoring.com/catalog/extractMethod.html).